webpush/encrypt.go

```diff
@@ -55,12 +55,36 @@ import (
 	"strings"
 )
 
+// Encryption version.
+type Version int
+
+const (
+	// The version that is widely deployed with WebPush (as of 2016-11).
+	AESGCM = iota
+	// The most recent version, the salt, record size and key identifier
+	// are included in a header that is part of the encrypted content coding.
+	AES128GCM
+)
+
+func (v Version) String() string {
+	switch v {
+	case AESGCM:
+		return "aesgcm"
+	case AES128GCM:
+		return "aes128gcm"
+	}
+	return ""
+}
+
 const (
-	maxPayloadLength = 4078
+	aesgcmMaxPayloadLength = 4078
+	// due to binary message header.
+	aes128gcmMaxPayloadLength = 4057
 )
 
 var (
 	authInfo = []byte("Content-Encoding: auth\x00")
+	keyInfoPrefix = []byte("WebPush: info\x00")
```

Comment:
It's in here: https://tools.ietf.org/html/draft-ietf-webpush-encryption-08#section-3.3

Added as a comment.